### PR TITLE
Oriented Box ROS2 converters

### PR DESCRIPTION
**Pull request by mikebauer**
## Description of change
Add ROS2 Converters for Oriented Boxes

## Guide to reproduce test results.
```bash
bazel test //resim/msg:oriented_box_from_ros2_test
bazel test //resim/transforms/proto:fuzz_helpers_test
bazel test //resim/geometry/proto:fuzz_helpers_test
bazel test //resim/msg:fuzz_helpers_test
```

## Checklist:
- [x] I have self-reviewed this change.
- [x] I have tested this change.
- [x] This change is covered by tests that are already landed, or in this PR.

### CODE REVIEW

File: resim/msg/fuzz_helpers.hh [hunk 1]
```diff
@@ -43,9 +45,16 @@ TransformStamped random_element(
 
   result.mutable_header()->CopyFrom(random_element<Header>(rng));
   result.set_child_frame_id(UUID::new_uuid().to_string());
-  const transforms::SE3 transform{transforms::SE3::exp(
-      testing::random_vector<transforms::SE3::TangentVector>(*rng))};
-  pack(transform, result.mutable_transform());
+  result.mutable_transform()->CopyFrom(
+      random_element<transforms::proto::SE3>(rng));
+
+  // We don't use these when converting to/from ROS2
```
Comment (knowlsie) [inline resim/msg/fuzz_helpers.hh:51]:
```suggestion
  // We don't use frame IDs when converting to/from ROS2.
```

File: resim/msg/fuzz_helpers.hh [hunk 2]
```diff
@@ -69,9 +78,14 @@ PoseWithCovariance random_element(
     TypeTag<PoseWithCovariance> /*unused*/,
     InOut<Rng> rng) {
   PoseWithCovariance result;
-  const transforms::SE3 pose{transforms::SE3::exp(
-      testing::random_vector<transforms::SE3::TangentVector>(*rng))};
-  pack(pose, result.mutable_pose());
+  result.mutable_pose()->CopyFrom(random_element<transforms::proto::SE3>(rng));
+
+  // We don't use these when converting to/from ROS2
```
Comment (knowlsie) [inline resim/msg/fuzz_helpers.hh:83]:
```suggestion
  // We don't use frame IDs when converting to/from ROS2.
```

File: resim/msg/oriented_box_from_ros2_test.cc [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+namespace resim::msg {
+
+TEST(OrientedBoxSE3FromRos2Test, TestRoundTrip) {
+  // SETUP
+  constexpr std::size_t SEED = 913U;
+  std::mt19937 rng{SEED};
+  geometry::proto::OrientedBoxSE3 test_oriented_box{
+      random_element<geometry::proto::OrientedBoxSE3>(InOut{rng})};
+  // We don't use these when converting to/from ROS2
```
Comment (knowlsie) [inline resim/msg/oriented_box_from_ros2_test.cc:27]:
```suggestion
  // We don't use frame IDs when converting to/from ROS2.
```

**Review by knowlsie — APPROVED** [review]
LGTM!

File: resim/geometry/proto/oriented_box.proto [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+syntax = "proto3";
+
+import "resim/transforms/proto/se3.proto";
+
+package resim.geometry.proto;
+
+message OrientedBoxSE3 {
+    resim.transforms.proto.SE3 reference_from_box = 1;
+    repeated double            extents            = 2;
```
Comment (knowlsie) [inline resim/geometry/proto/oriented_box.proto:15]: Not sure how to consistently label units, or if we should.

File: resim/msg/fuzz_helpers.hh (hunk 1 above)
Comment (mikebauer) [reply resim/msg/fuzz_helpers.hh]: Good call. Definitely clearer this way.

File: resim/geometry/proto/oriented_box.proto (hunk 4 above)
Comment (mikebauer) [reply resim/geometry/proto/oriented_box.proto:15]: In this case, I went with this to match the fields in geometry::OrientedBox exactly. I think the idea of an oriented box shouldn't necessarily be opinionated about units, but at some level (maybe detections in this case) we'll have to be?

**Pull request merged**